refined/src/main/scala/io/finch/refined/PredicateFailed.scala

```diff
@@ -0,0 +1,9 @@
+package io.finch.refined
+
+import scala.util.control.NoStackTrace
+
+case class PredicateFailed(error: String) extends Exception with NoStackTrace {
```

Comment:
Maybe we can throw `Error.NotValid` for now? We can supply it with a generic `RequestItem` for now.

Reply:
That was actually a problem as I had no `RequestItem` in scope.

I think it's fine as it is now, because `param` and other endpoints will wrap it into `Error.NotParsed`:
https://github.com/finagle/finch/blob/master/core/src/main/scala/io/finch/endpoint/param.scala#L95

Reply:
Oh I see. That makes sense!